src/node_buffer.cc

```diff
@@ -728,6 +728,10 @@ void StringWrite(const FunctionCallbackInfo<Value>& args) {
   if (max_length == 0)
     return args.GetReturnValue().Set(0);
 
+  // Since V8 restrict capacity in String::WriteUtf8 within the int range
+  if (max_length > std::numeric_limits<int>::max())
```

Comment:
I think this would just result in lossy writes? IMO we should just throw `ERR_STRING_TOO_LONG` when the estimated size is bigger than `v8::String::kMaxLength` in `StringBytes::Write`.

Reply:
thanks @joyeecheung , I have update this and add judgement in `StringWrite` instead of `StringBytes::Write`, because I saw some place will pass a negative value as `buflen`'s argument to `StringBytes::Write` which maybe bring inconsistent。 I have also opened an issue on #51830 to describe it

Reply:
IMO this was fine before and now it's a semver major change since it can cause code which didn't throw before to now throw. [The documentation states](https://nodejs.org/api/buffer.html#bufwritestring-offset-length-encoding):

> length [<integer>](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#Number_type) Maximum number of bytes to write (written bytes will not exceed buf.length - offset). Default: buf.length - offset.

To me "maximum number of bytes to write" indicates that it's acceptable to pass a large number as well.

There's no way to cause a lossy write if you just set `max_length = std::numeric_limits<int>::max()`, since `v8::String::kMaxLength` is an `int` anyway.

Reply:
IMO one potential benefit of not throwing an exception when the buffer's `size` is greater than `v8::String::kMaxLength `and `length` is not specified for` buffer.write(str, offset, length)` might reduce confusion for users?